### PR TITLE
AWS Athena ODBC driver support for App Access

### PR DESCRIPTION
**Pull request by greedy52**
Implements #8281
- #8281

Note that both Athena JDBC driver and Timestream JDBC driver use Java SDK v1 so Teleport would not able to support them (details see https://github.com/gravitational/teleport/issues/19366)

## Changes
Interestingly, the copy header logic change from https://github.com/gravitational/teleport/pull/19387 actually fixed an issue that AWS would reject ODBC driver requests when `Expect: 100-continue` headers are signed (by the old logic). I had a similar change prior to Gavin's refactor but it's no longer necessary.

This PR then adds a new `athena-odbc` format `tsh proxy aws` to dump pre-compiled list of ODBC properties.

## UX
Login
```
$ tsh apps login aws
Logged into AWS app "aws".

Your IAM role:
  arn:aws:iam::<account_account_id>:role/steve-poweruser

Example AWS CLI command:
  tsh aws s3 ls

Or start a local proxy:
  tsh proxy aws --app aws
```
New proxy format
```
$ tsh proxy aws --app aws -p 8888 -f athena-odbc
Started AWS proxy on http://127.0.0.1:8888.

Set the following properties for the Athena ODBC data source:
[Teleport Athena Access]
AuthenticationType = IAM Credentials
UID = <aws_access_key_id>
PWD = <aws_secret_access_key>
UseProxy = 1;
ProxyScheme = http;
ProxyHost = 127.0.0.1;
ProxyPort = 8888;
TrustedCerts = <path_to_ca_bundle>

Here is a sample connection string using the above credentials and proxy settings:
DRIVER=Simba Amazon Athena ODBC Connector;AwsRegion=us-east-1;S3OutputLocation=s3://example-bucket/athena/output/;Workgroup=example-workgroup;AuthenticationType=IAM Credentials;UID=<aws_access_key_id>;PWD=<aws_secret_access_key>;UseProxy=1;ProxyScheme=http;ProxyHost=127.0.0.1;ProxyPort=8888;TrustedCerts=<path_to_ca_bundle>
```

### CODE REVIEW

File: tool/tsh/proxy.go [hunk 1]
```diff
@@ -818,19 +830,36 @@ const (
...
-		"Optional format to print the commands for setting environment variables, one of: %s.",
+		"Optional format to print the commands for setting environment variables, one of: %s. Default is %s.",
 		strings.Join(envVarFormats, ", "),
+		envVarDefaultFormat(),
+	)
+}
+
+func awsProxyFormatFlagDescription() string {
+	return fmt.Sprintf(
+		"%s Or speicfy a service format, one of: %s",
```
Comment (GavinFrazar) [inline tool/tsh/proxy.go:860]:
```suggestion
		"%s Or specify a service format, one of: %s",
```

**Review by GavinFrazar — APPROVED** [review]
LGTM, glad the signing service refactor fixed a header signing issue I wasn't even aware of 😄

**Review by mdwn — APPROVED** [review]
LGTM!

File: tool/tsh/proxy.go [hunk 2]
```diff
@@ -911,3 +940,24 @@ In addition to the endpoint URL, use the following credentials to connect to the
...
 `))
+
+// awsProxyAthenaODBCTemplate is the message that gets printed to a user when an
+// AWS proxy is used for Athena ODBC driver.
+var awsProxyAthenaODBCTemplate = template.Must(template.New("").Funcs(awsTemplateFuncs).Parse(
+	`Started AWS proxy on {{.envVars.HTTPS_PROXY}}.
+{{if .randomPort}}To avoid port randomization, you can choose the listening port using the --port flag.
+{{end}}
+Set the following properties for the Athena ODBC data source:
+[Teleport Athena Access]
```
Comment (codingllama) [inline tool/tsh/proxy.go:951]: Maybe?

```suggestion
[Teleport AWS Athena Access]
```

File: tool/tsh/proxy.go [hunk 3]
```diff
@@ -911,3 +940,24 @@ In addition to the endpoint URL, use the following credentials to connect to the
...
+UID = {{.envVars.AWS_ACCESS_KEY_ID}}
+PWD = {{.envVars.AWS_SECRET_ACCESS_KEY}}
+UseProxy = 1;
+ProxyScheme = {{.proxyScheme}};
+ProxyHost = {{.proxyHost}};
+ProxyPort = {{.proxyPort}};
+TrustedCerts = {{.envVars.AWS_CA_BUNDLE}}
+
+Here is a sample connection string using the above credentials and proxy settings:
+DRIVER=Simba Amazon Athena ODBC Connector;AwsRegion=us-east-1;S3OutputLocation=s3://example-bucket/athena/output/;Workgroup=example-workgroup;AuthenticationType=IAM Credentials;UID={{.envVars.AWS_ACCESS_KEY_ID}};PWD={{.envVars.AWS_SECRET_ACCESS_KEY}};UseProxy=1;ProxyScheme={{.proxyScheme}};ProxyHost={{.proxyHost}};ProxyPort={{.proxyPort}};TrustedCerts={{.envVars.AWS_CA_BUNDLE}}
```
Comment (codingllama) [inline tool/tsh/proxy.go:962]: Could we print a placeholder for AWS_SECRET_ACCESS_KEY instead?

**Review by codingllama — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

File: tool/tsh/proxy.go (hunk 3 above)
Comment (greedy52) [reply tool/tsh/proxy.go:975]: The actual value of AWS_SECRET_ACCESS_KEY must be presented here. It's generated by tsh, and it's used by the client to sign the requests, so we can verify the source is true.

**Pull request merged**